testing/test_package_experiments/lib/constructor_tearoffs.dart

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Exercise a number of constructor-tearoff related features inside dartdoc.
+///
+/// References to tearoffs should work at a top level too:
+///   [A.new], [B.new], [At.new], [Bt.new], [C], [C.new], [D.new],
+///
+library constructor_tearoffs;
+
+abstract class A {
+  final int number;
+  /// Even though this is abstract, dartdoc should still allow referring to
+  /// [A.new].
+  A.new(this.number);
+}
+
+class B extends A {
+  B.new() : super(5);
+}
+
+class C {}
+
+typedef At = A;
+typedef Bt = B;
+typedef Ct = C;
+
+/// [Dt.new] should be a thing.
```

Comment:
Typo, I think: `[D.new]`

But this is interesting. It is illegal to tearoff `D.new`, since it is abstract (it cannot be directly called). But folk may still want to refer to it. Because of this, maybe `[D.new]` should be left as a PrefixedIdentifier, rather than a ConstructorReference, in the analyzer's AST...

Reply:
Both are intended to work, I'll add the other. 

As for the AST implications, I'm guessing it might be a good idea there, yes, for comment reference resolution as well as possibly dealing with erroneous code more cleanly?  I don't know the extent to which this is important for those cases.